cpp/src/arrow/array/builder_nested.cc

```diff
@@ -213,13 +213,13 @@ Status FixedSizeListBuilder::AppendValues(int64_t length, const uint8_t* valid_b
 Status FixedSizeListBuilder::AppendNull() {
   RETURN_NOT_OK(Reserve(1));
   UnsafeAppendToBitmap(false);
-  return value_builder_->AppendNulls(list_size_);
+  return value_builder_->AppendEmptyValues(list_size_);
```

Comment:
So, when child is non-nullable, would it matter previously?

Reply:
More details in the issue https://github.com/apache/arrow/issues/41188

Summary:

When you append nulls you force the creation of a null bitmap for the entire array.

`[[1, 2], null, [3, 4]]` currently needs a `[1, 0 1]` validity map at the top-level and a `[1, 1, 0, 0, 1, 1]` map in the child values array when it could have all values as valid and an empty bitmap.

Reply:
But when you have an array like `[[1, null], null, [3, 4]]`, then you cannot run away from having a validity bitmap for the child values array —— `[1, 0, , 1, 1, 1, 1]`.

Reply:
So might be a performance for deep nested thing like `FixLengthArray<struct<str, str, str>, ...>`?

Reply:
It reduces the number of scattered bit reads and writes when dealing with fixed-size-lists.